pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -16,29 +15,13 @@ type Bootstrap struct {
 
 var _ asset.WritableAsset = (*Bootstrap)(nil)
 
-// Dependencies returns the assets on which the Bootstrap asset depends.
-func (a *Bootstrap) Dependencies() []asset.Asset {
-	return append(
-		a.Common.Dependencies(),
-		&CVOIgnore{},
-	)
-}
-
 // Generate generates the ignition config for the Bootstrap asset.
 func (a *Bootstrap) Generate(dependencies asset.Parents) error {
 	templateData := a.getTemplateData(dependencies, false)
 	if err := a.generateConfig(dependencies, templateData); err != nil {
 		return err
 	}
 
-	// replace cvo-overrides.yaml so that CVO does not try to create resources for the manifests that the installer
-	// is supplying to cluster-bootstrap
-	cvoIgnore := &CVOIgnore{}
-	dependencies.Get(cvoIgnore)
-	for _, file := range ignition.FilesFromAsset(rootDir, "root", 0644, cvoIgnore) {
-		a.Config.Storage.Files = replaceOrAppend(a.Config.Storage.Files, file)
-	}
```

Comment:
Hm... I understand that this code was introduced in #5271 and that PR was created to skip cvo overrides for bootstrap in place, but it is unclear to me how this specific code relates to bootstrap in place. I can spend some more time on trying to grok this tomorrow, but I wanted to mention it in case it is obvious to anyone else and they might be able to educate me.

At first blush, it seems to me like we might still need this code and only the changes in bootkube.sh need to be addressed. I hope my question/observation makes sense.

Reply:
You are right, we still need it, it just moved to [pkg/asset/ignition/bootstrap/common.go](https://github.com/openshift/installer/pull/6649/files#diff-2a27cf9f710e504a0f8b3710fa10d24e403ff92e781cc8acbb425571427cc51bR518).
It seems that now bootstrap/bootstrap.go has nothing to add on top of bootstrap/common.go and we can do some refactoring (delete bootstrap.go, rename common.go to bootstrap.go and `Common` to `Bootstrap`, and embedding `Bootstrap` in `SingleNodeBootstrapInPlace` (instead of Common), that might make things clearer but will require the reverse refactoring the next time bootstrap and bootstrap in place diverge

Reply:
Thanks for clarifying that common.go already had the equivalent code.